test/built-ins/Date/prototype/toTemporalInstant/this-value-valid-date.js

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2024 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-date.prototype.totemporalinstant
+description: >
+  Return value for valid dates.
+info: |
+  Date.prototype.toTemporalInstant ( )
+
+  ...
+  3. Let t be dateObject.[[DateValue]].
+  4. Let ns be ? NumberToBigInt(t) × ℤ(10**6).
+  5. Return ! CreateTemporalInstant(ns).
+features: [Temporal, BigInt]
+---*/
+
+assert.sameValue(
+  new Date(0).toTemporalInstant().epochNanoseconds,
+  0n,
+  "start of the epoch"
+);
+
+assert.sameValue(
+  new Date(123_456_789).toTemporalInstant().epochNanoseconds,
+  123_456_789_000_000n,
+  "date after start of the epoch"
+);
+
+assert.sameValue(
+  new Date(-123_456_789).toTemporalInstant().epochNanoseconds,
+  -123_456_789_000_000n,
+  "date before start of the epoch"
```

Comment:
Nit: the epoch is a point in time, so "start of the epoch" should be "the (Unix) epoch" (three times)